admin_guide/encrypting_data.adoc

```diff
@@ -0,0 +1,261 @@
+is returned which prevents clients from accessing that resource.
+
+[IMPORTANT]
+====
+If any resource is not readable via the encryption config (because keys were changed),
+the only recourse is to delete that key from the underlying etcd directly. Calls that attempt to
+read that resource will fail until it is deleted or a valid decryption key is provided.
+====
+
+[[encrypting-data-providers]]
+=== Available Providers
+
+[cols="1,1,1,1,1,3"]
+|===
+| Name | Encryption | Strength | Speed | Key Length | Other Considerations
+
+| *identity*
+| None
+| N/A
+| N/A
+| N/A
+| Resources written as-is without encryption. When set as the first provider, the resource will be
+decrypted as new values are written.
+
+| *aescbc*
+| AES-CBC with PKCS#7 padding
+| Strongest
+| Fast
+| 32-byte
+| The recommended choice for encryption at rest but may be slightly slower than *secretbox*.
+
+| *secretbox*
+| XSalsa20 and Poly1305
+| Strong
+| Faster
+| 32-byte
+| A newer standard and may not be considered acceptable in environments that require high levels of
+review.
+
+| *aesgcm*
+| AES-GCM with random nonce
+| Must be rotated every 200k writes
+| Fastest
+| 16, 24, or 32-byte
+| Is not recommended for use except when an automated key rotation scheme is implemented.
+
+|===
+
+Each provider supports multiple keys - the keys are tried in order for decryption, and if the
+provider is the first provider, the first key is used for encryption.
+
+[[encrypting-data-process]]
+== Encrypting Data
+
+Create a new encryption config file
+
+[source,yaml]
+----
+kind: EncryptionConfig
+apiVersion: v1
+resources:
+  - resources:
+    - secrets
+    providers:
+    - aescbc:
+        keys:
+        - name: key1
+          secret: <BASE 64 ENCODED SECRET>
+    - identity: {}
+----
+
+To create a new secret perform the following steps:
+
+. Generate a 32 byte random key and base64 encode it. If you are on Linux or Mac OS X, run the
+following command:
++
+----
+head -c 32 /dev/urandom | base64
+----
+
+. Place that value in the *secret* field
+
+. Restart API server
+
+[IMPORTANT]
+====
+Config file contains keys that can decrypt content in etcd, so you must properly restrict
+permissions on masters so only the user who runs the master API server can read it.
+====
+
+
+[[encrypting-data-verification]]
+== Verifying that Data is Encrypted
+
+Data is encrypted when written to etcd. After restarting API server any newly created or
+updated secret should be encrypted when stored. To check, you can use the `etcdctl` command line
+program to retrieve the contents of your secret.
+
+. Create a new secret called `secret1` in the `default` namespace:
++
+----
+oc create secret generic secret1 -n default --from-literal=mykey=mydata
+----
+
+. Using the etcdctl commandline, read that secret out of etcd:
++
+----
+ETCDCTL_API=3 etcdctl get /kubernetes.io/secrets/default/secret1 [...] | hexdump -C
+----
++
+where *[...]* must be the additional arguments for connecting to the etcd server.
```

Comment:
That's very vague. It would be nice to point to an appropriate document showing what the other flags are. The only one I found is this: https://docs.openshift.org/latest/admin_guide/backup_restore.html